src/IK/include/BipedalLocomotion/IK/DistanceTask.h

```diff
@@ -0,0 +1,149 @@
+     * |:----------------------------------:|:----------------------------:|:----------------------------------------------------------------------------------------------------------------------------------------:|:---------:|
+     * |   `robot_velocity_variable_name`   |            `string`          |                            Name of the variable contained in `VariablesHandler` describing the robot velocity                            |    Yes    |
+     * |                `kp`                |            `double`          |                                               Gain of the distance controller                                                            |    Yes    |
+     * |         `target_frame_name`        |            `string`          |                                       Name of the frame of which computing the distance                                                  |    Yes    |
+     * |          `base_frame_name`         |            `string`          |          Name of the frame with respect to which computing the distance. If empty, the world frame will be used (Default = "")           |    No     |
+     * @return True in case of success, false otherwise.
+     */
+    // clang-format on
+    bool
+    initialize(std::weak_ptr<const ParametersHandler::IParametersHandler> paramHandler) override;
+
+    /**
+     * Set the kinDynComputations object.
+     * @param kinDyn pointer to a kinDynComputations object.
+     * @return True in case of success, false otherwise.
+     */
+    bool setKinDyn(std::shared_ptr<iDynTree::KinDynComputations> kinDyn) override;
+
+    /**
+     * Set the set of variables required by the task. The variables are stored in the
+     * System::VariablesHandler.
+     * @param variablesHandler reference to a variables handler.
+     * @note The handler must contain a variable named as the parameter
+     * `robot_velocity_variable_name` stored in the parameter handler. The variable represents the
+     * generalized velocity of the robot. Where the generalized robot velocity is a vector
+     * containing the base spatial-velocity (expressed in mixed representation) and the joints
+     * velocity.
+     * @return True in case of success, false otherwise.
+     */
+    bool setVariablesHandler(const System::VariablesHandler& variablesHandler) override;
+
+    /**
+     * Update the content of the task.
+     * @return True in case of success, false otherwise.
+     */
+    bool update() override;
+
+    /**
+     * @brief setDesiredDistance
+     * @param desiredDistance The desired distance expressed in meters
+     * @note The desired distance is considered without sign.
+     * @return True in case of success, false otherwise.
+     */
+    bool setDesiredDistance(double desiredDistance);
```

Comment:
Can you rename it as `setSetPoint()` as the other tasks?

Reply:
I have added the method in https://github.com/ami-iit/bipedal-locomotion-framework/pull/717/commits/f86f2a2394b6de040e88e85c037c9dd1af035527. 

I preferred to keep also the other method since it is a bit more informative of a name